tests/ui/methods/method-self-arg.rs

```diff
@@ -9,6 +9,7 @@ struct Foo;
 impl Foo {
     fn foo(self, x: &Foo) {
         unsafe { COUNT *= 2; }
+        //~^ WARN creating a mutable reference to mutable static is discouraged [static_mut_refs]
```

Comment:
Since this test is unrelated, let's fix (or `allow(..)` and mark `FIXME`) this and other similar uses instead.  This could be fixed, e.g., with `COUNT = COUNT * 2`.